### DISCUSSION TITLE
How to pass data into the action method?

### DISCUSSION THREAD

**Discussion by bilogic** [filamentphp/filament#8775] [Help]
```PHP
$actions[] = \Filament\Actions\Action::make('Test')
   ->label(__('Test'))
   ->color('gray')
   ->action('performTest')
   ->arguments(['option1']);

...

public function performTest(Request $request)
{
    // this method is called after clicking on the `Test` action button
    
    // 1. how do i read `arguments` here? or how do I pass something here?
    // 2. how do i refresh the whole page and also show a notification?
}
```

**Comment by lourbert** [comment]
1. To read the arguments passed to the `performTest()` action method, you can use the` $request->arguments->get()` method. This will return an array of all the arguments that were passed to the action method. In your case, the array will contain the following value:

```php
['option1' => 'value']
```

2. To refresh the whole page and also show a notification, you can use the following code:

```php
$this->notify('success', 'Your test was successful!');
return redirect()->route('filament.resources.filament-resource.index');
```
The notify() method will show a notification to the user, and the redirect() method will redirect the user to the specified route. In your case, the user will be redirected to the index page of the Filament resource.

Here is a complete example of the performTest() action method:
```php
public function performTest(Request $request)
{
    $option1 = $request->arguments->get('option1');

    // Do something with the option1 variable

    $this->notify('success', 'Your test was successful!');
    return redirect()->route('filament.resources.filament-resource.index');
}

**Reply by bilogic** [reply]
@lourbert 

Thanks but I'm getting
```PHP
        // Method App\Filament\Pages\Test::notify does not exist.
        $this->notify('success', 'Your test was successful!');
        
        // Call to a member function get() on null
        dd($request->arguments->get());         
```

I have written this Gist in a way that one should be able to copy and paste into their filament project, the only required change is the namespace.

Would be most grateful if you could help me figure out this mystery https://gist.github.com/bilogic/3c8d9bb37e3a81e319fb2e5b8b8d364a#file-testpage-php

Thank you.
